src/cargo/util/command_prelude.rs

```diff
@@ -1106,6 +1110,56 @@ pub fn get_registry_candidates() -> CargoResult<Vec<clap_complete::CompletionCan
     }
 }
 
+fn get_profile_candidates() -> CargoResult<Vec<clap_complete::CompletionCandidate>> {
+    match get_workspace_profile_candidates() {
+        Ok(candidates) if !candidates.is_empty() => Ok(candidates),
+        // fallback to default profile candidates
+        _ => Ok(default_profile_candidates()),
+    }
+}
+
+fn get_workspace_profile_candidates() -> CargoResult<Vec<clap_complete::CompletionCandidate>> {
+    let gctx = new_gctx_for_completions()?;
+    let ws = Workspace::new(&find_root_manifest_for_wd(gctx.cwd())?, &gctx)?;
+    let profiles = Profiles::new(&ws, InternedString::new("dev"))?;
+
+    let mut candidates = Vec::new();
+    for name in profiles.profile_names() {
+        if let Ok(profile_instance) = Profiles::new(&ws, name) {
```

Comment:
We are recreating `Profiles` struct here, which doesn't really make sense, as `Profiles` already has every profile. But for completion, this seems acceptable?

BTW we could use let-else so one less identation level.

```suggestion
        let Ok(profile_instance) = Profiles::new(&ws, name) else {
            continue;
        };
```

Reply:
> BTW we could use let-else so one less identation level.

done 

> We are recreating Profiles struct here, which doesn't really make sense, as Profiles already has every profile. But for completion, this seems acceptable?

you are right that recreating the `Profiles` struct for each profile name is inefficient, the original profiles object should already contain all the information, I kept this approach for consistency with the existing implementation, but if you think it would be better to refactor this to avoid the redundant Profiles creation i would be happy to refactor it